docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+
+```bash
+/ # nc -zv prometheus-rsync.NAMESPACE.svc.cluster.local 873
+prometheus-rsync.NAMESPACE.svc.cluster.local (10.130.32.5:873) open
+```
+
+## Modifying the `kubeprod-manifest.jsonnet` to add the sidecar rsync container
+
+The `kubeprod-manifest.jsonnet` file will contain all the custom configuration you want to apply to the jsonnet manifests.  In this case, in order to add the rsync sidecar container the following configuration will have to be added:
+
+```jsonnet
+// Bitnami specific common configuration
+local common_prom_config = import "../common/prometheus/conf.d/prometheus.jsonnet";
+
+// Common libs
+local kube = import "kube.libsonnet";
+
+(import "../vendor/kubeprod/manifests/platforms/gke.jsonnet") {
+
+  config:: import "kubeprod-autogen.json",
+
+  // Place your overrides here
+  prometheus+: { 
+
+    // Add rsync sidecar container to the prometheus sts.
+    // and adapt resources to our needs
+    prometheus+: {  // Prometheus server itself
+      deploy+: {  // Prometheus StatefulSet deployment
+        spec+: {
+          template+: {
+            spec+: {
+              containers_+: {
+                rsync: kube.Container("sidecar-rsync") {
+                  image: "alpine:3.8",
+                  args+: ["sh", "-c", "apk update && apk add rsync curl && tail -f /dev/null"],
+                  volumeMounts_+: {
+                    data: { mountPath: "/data" },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+}
+```
+
+After you deploy the manifests in your Kubernetes cluster, you will see the 3 containers that the Prometheus pod has:
+
+```bash
+~ $ kubectl get pods -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}' | grep prometheus |\
+sort
+
+prometheus-0:	bitnami/prometheus:2.4.3-r31, jimmidyson/configmap-reload:v0.2.2, alpine:3.8,
+```
+
+Now, you can do a quick test of the rsync service by getting a shell in the container executing:
+
+`kubectl exec -it prometheus-0 -n kubeprod -c sidecar-rsync sh`
+
+and:
+
+`rsync -avrP prometheus-rsync.NAMESPACE.svc.cluster.local::data/test-file .`
+
+> Make sure previously you have deployed a test file in your Prometheus data directory so you can rsync it to the new container
+
+## Syncing with the compact times of the Prometheus database
+Now that all bits are in place you are ready to perform the Prometheus TSDB migration. To do so, you will first have to sync with the TSDB compact times of Prometheus to ensure the minimal data loss.
+
+### Doing a Prometheus TSDB migration
+
+By default, each two hours Prometheus will perform a compact of the TSDB and will write into disk the latest information it has in the WAL (Write-Ahead-Log) file. You can notice about these compacts in the TSDB by looking at the Prometheus logs:
+
+```bash
```

Comment:
At first reading, I thought these were commands to read the logs because the previous line says "by looking at the Prometheus logs". I then realized this was actually log output. I would change the previous to make the code block clearer ie. "You can watch the compaction process through the Prometheus logs. A small sample of the relevant log lines is shown below"

Reply:
done, thank you.